src/core/lib/surface/call.cc

```diff
@@ -694,6 +694,10 @@ static void cancel_with_error(grpc_call* c, grpc_error* error) {
   execute_batch(c, op, &state->start_batch);
 }
 
+void grpc_call_cancel_and_execute(grpc_call* call) {
```

Comment:
Instead of `grpc_call_cancel_and_execute`, name this `grpc_call_cancel_internal`. And put it in an internal header, rather than a surface header? I think this could be similar to [slice_ref_internal.h](https://github.com/grpc/grpc/blob/master/src/core/lib/slice/slice_internal.h#L28)

Reply:
Done.

Reply:
Done.